Atomic/sign.py

```diff
@@ -0,0 +1,107 @@
+from . import util
+from . import Atomic
+import os
+import tempfile
+from .atomic import AtomicError
+import re
+
+
+class Sign(Atomic):
+    def sign(self):
+        # TODO
+        # Atomic is run as sudo. Should we work around that?
+
+        if self.args.debug:
+            util.write_out(str(self.args))
+
+        # Verify all the images provided are valid
+        self.check_input_validity()
+
+        signature_path = util.get_signature_path(self.atomic_config)
+        use_def_sig_path = True if signature_path == self.args.output else False
```

Comment:
This is confusing. The sigpath mechanism (creating a directory structure based on reference) and the `-o` mechanism (writing to the named file) use the input string in a completely different way to arrive at the final output location.

`-o /var/lib/atomic/containers` should not behave as if the option is not there (it should probably fail with “trying to overwrite a directory with a file”).

AFAICS this variable should be removed, and its user replaced by simple `if self.args.output is None` (which would, in turn require removing the default value in the argument parser.).